File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/jdbc4/JDBC4PooledConnection.java
package io.roastedroot.sqlite4j.jdbc4;

import javax.sql.PooledConnection;
import javax.sql.StatementEventListener;
import org.jspecify.annotations.NonNull;

public abstract class JDBC4PooledConnection implements PooledConnection {

    public void addStatementEventListener(@NonNull StatementEventListener listener) {
        // TODO impl
    }

    public void removeStatementEventListener(@NonNull StatementEventListener listener) {
        // TODO impl
    }
}

File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/jdbc4/JDBC4PooledConnection.kt
package io.roastedroot.sqlite4j.jdbc4

import javax.sql.PooledConnection
import javax.sql.StatementEventListener

abstract class JDBC4PooledConnection : PooledConnection {
    override fun addStatementEventListener(listener: StatementEventListener) {
        // TODO impl
    }

    override fun removeStatementEventListener(listener: StatementEventListener) {
        // TODO impl
    }
}
